compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/BIRGen.java

```diff
@@ -698,6 +698,7 @@ public void visit(BLangFunction astFunc) {
         BIRBasicBlock enclBB = this.env.enclBB;
         if (enclBB.instructions.size() == 0 && enclBB.terminator == null && this.env.returnBB != null) {
             enclBB.terminator = new BIRTerminator.GOTO(null, this.env.returnBB);
+            setScope(enclBB.terminator);
```

Comment:
Shall we move the setScope inside constructor. We need to pass the current scope. Then we do not need to duplicate the code.

Reply:
We need to do for all terminators which can have scope..

Reply:
Addressed with https://github.com/ballerina-platform/ballerina-lang/pull/31829/commits/111094729f69d3296a3c694581dabfdebc283d93